### PR TITLE
lora/lorawan: implementation of LoRaWAN stack

### PR DESCRIPTION
**Pull request by deadprogram**
This PR basically starts from the @ofauchon implementation in https://github.com/ofauchon/go-lorawan-stack but refactored.

Already able to join The Things Network using the atcmd example!

### CODE REVIEW

**Comment by ofauchon** [comment]
Thank you @deadprogram for this awesome PR !

I'm glad that my "very basic" Lorawan Stack is now part of TinyGo drivers. 

Now we can go ahead and work with the community to add all the missing features like: 

* Lorawan MAC Commands
* Regional Frequency Parameters
* Downlink messages

TinyGo will be soon a great alternative for IoT and connected objects. 

Olivier

File: lora/radio.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	Reset()
+	Tx(pkt []uint8, timeoutMs uint32) error
+	Rx(timeoutMs uint32) ([]uint8, error)
+	SetFrequency(freq uint32)
+	SetIqMode(mode uint8)
+	SetCodingRate(cr uint8)
+	SetBandwidth(bw uint8)
+	SetCrc(enable bool)
+	SetSpreadingFactor(sf uint8)
+}
```
Comment (ofauchon) [inline lora/radio.go:13]: We could add LoraConfig() , SetSyncWord , SetTxPower, SetPreamble ... 
as well ....

File: examples/lora/lorawan/atcmd/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+## PyBadge with LoRa Featherwing 
+
+```
+tinygo flash -target pybadge -tags featherwing ./examples/lora/atcmd/
+```
+
+## LoRa-E5 
+
+```
+tinygo flash -target lorae5 ./examples/lora/atcmd/
```
Comment (ofauchon) [inline examples/lora/lorawan/atcmd/README.md]: Path to "atcmd" is examples/lora/lorawan/atcmd

**Review by ofauchon — COMMENTED** [review]
I built and flashed the code on LoraE5 dev board (stm32wl/sx126x) 
 
-> No problem running 'atcmd' example. 
-> I could join my private lorawan gateway (Chirpstack) with it.

Looks good to me.

File: examples/lora/lorawan/atcmd/README.md (hunk 2 above)
Comment (deadprogram) [reply examples/lora/lorawan/atcmd/README.md]: Corrected.

File: lora/radio.go (hunk 1 above)
Comment (deadprogram) [reply lora/radio.go:13]: We can add in future PRs as needed.

**Pull request merged**